src/query/combine.rs

```diff
@@ -42,10 +45,17 @@ where
                 None => {
                     let col = match &sel.expr {
                         SimpleExpr::Column(col_ref) => match &col_ref {
-                            ColumnRef::Column(col) => col,
-                            ColumnRef::TableColumn(_, col) => col,
+                            ColumnRef::Column(col) | ColumnRef::TableColumn(_, col) => col,
+                        },
+                        SimpleExpr::AsEnum(_, simple_expr) => match simple_expr.as_ref() {
+                            SimpleExpr::Column(col_ref) => match &col_ref {
+                                ColumnRef::Column(col) | ColumnRef::TableColumn(_, col) => col,
+                            },
+                            _ => {
+                                panic!("cannot apply alias for AsEnum with expr other than Column")
+                            }
                         },
-                        _ => panic!("cannot apply alias for expr other than Column"),
```

Comment:
It panic at this line. It didn't catch `ColumnRef::AsEnum` and apply alias for it.

Reply:
@tyt2y3